presto-docs/src/main/sphinx/connector/bigquery.rst

```diff
@@ -0,0 +1,156 @@
+===================================== ============================================================== ==============================================
+bigquery.project-id                   The Google Cloud Project ID where the data reside              Taken from the service account
+bigquery.parent-project               The project ID Google Cloud Project to bill for the export     Taken from the service account
+bigquery.parallelism                  The number of partitions to split the data into                The number of executors
+bigquery.views-enabled                Enables the connector to read from views and not only tables.  ``false``
+                                      Please read `this section <#reading-from-views>`_ before
+                                      enabling this feature.
+bigquery.view-materialization-project The project where the materialized view is going to be created The view's project
+bigquery.view-materialization-dataset The dataset where the materialized view is going to be created The view's dataset
+bigquery.max-read-rows-retries        The number of retries in case of retryable server issues       ``3``
+bigquery.credentials-key              The base64 encoded credentials key                             None. See `authentication <#authentication>`_
+bigquery.credentials-file             The path to the JSON credentials file                          None. See `authentication <#authentication>`_
+===================================== ============================================================== ==============================================
+
+Data Types
+----------
+
+With a few exceptions, all BigQuery types are mapped directly to their Presto counterparts.
+Here are all the mappings:
+
+=========  ======================== =============================================================================================================
+BigQuery   Presto                   Notes
+=========  ======================== =============================================================================================================
+BOOLEAN    BOOLEAN
+BYTES      VARBINARY
+DATE       DATE
+DATETIME   VARCHAR                  In "yyyy-MM-dd'T'HH:mm:ss.SSSSS" format
+FLOAT      DOUBLE
+GEOGRAPHY  VARCHAR                  In `Well-known text (WKT) <https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry>`_ format
+INTEGER    BIGINT
+NUMERIC    DECIMAL(38,9)
+RECORD     ROW
+STRING     VARCHAR
+TIME       TIME_WITH_TIME_ZONE      Time zone is UTC
+TIMESTAMP  TIMESTAMP_WITH_TIME_ZONE Time zone is UTC
+=========  ======================== =============================================================================================================
+
+FAQ
+---
+
+What is the Pricing for the Storage API?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+See the `BigQuery pricing documentation <https://cloud.google.com/bigquery/pricing#storage-api>`_.
+
+Calling count(*) takes a long time
```

Comment:
@electrum @findepi Perhaps I'm missing something, but currently `SELECT * FROM t` and `SELECT COUNT(*) FROM t` looks the same where I create the API query, as `applyProjection()` was not called in both cases. How do I distinct an empty row from a full row? I there some call that I have missed?

Reply:
You get the list of required columns also in `io.prestosql.split.PageSourceProvider#createPageSource`.

Reply:
Unfortunately it is too late for me. The way the BIgQuery Storage API works is that I first open a `ReadSession`, where I need to specify the required columns. In response I get a list of streams and each split read from those streams. While I can create empty rows in the PageSource based on this, it will be only after the data has been read from the API.

Currently I create the read session in the `ConnectoeSpiltManager`, where I don't have this information. The only workaround I found so far is that I've noticed that `ConnectorMetadata.getColumnMetadata()` is called for each requested column, and use this  somehow using the session or the `ConnectorTableHandle`

Reply:
> currently `SELECT * FROM t` and `SELECT COUNT(*) FROM t` looks the same where I create the API query, as `applyProjection()` was not called in both cases.

@davidrabinowitz  is this reproducible with eg a postgres connector? can you file an issue?

Reply:
Sorry, couldn't find it, as the postgres connector relies on the base-jdbc plugin. It's `getSplits()` relies on the default implementation, and there is no `ConnectorPageSourceProvider`

I've submitted Issue #2943 per your suggestion

Reply:
I commented on the issue. `applyProjection()` seems to be called as expected with an empty projection list for this case.

Reply:
Thanks for the tip, it is implemented now by running count(*) on BigQuery